src/main/scala/edu/ie3/osmogrid/io/input/InputDataProvider.scala

```diff
@@ -6,73 +6,145 @@
 
 package edu.ie3.osmogrid.io.input
 
-import akka.actor.typed.Behavior
-import akka.actor.typed.ActorRef
-import akka.actor.typed.PostStop
-import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.{ActorContext, Behaviors, StashBuffer}
+import akka.actor.typed.{ActorRef, Behavior}
+import com.acervera.osm4scala.EntityIterator.fromPbf
+import com.acervera.osm4scala.model.{NodeEntity, RelationEntity, WayEntity}
 import edu.ie3.datamodel.models.input.connector.`type`.{
   LineTypeInput,
   Transformer2WTypeInput
 }
 import edu.ie3.osmogrid.ActorStopSupport
 import edu.ie3.osmogrid.cfg.OsmoGridConfig
-import edu.ie3.osmogrid.guardian.OsmoGridGuardian
-import edu.ie3.osmogrid.io.input.InputDataProvider.Request
-import edu.ie3.osmogrid.lv.coordinator.LvCoordinator.stopBehavior
-import edu.ie3.osmogrid.model.{OsmoGridModel, PbfFilter}
-import org.slf4j.Logger
+import edu.ie3.osmogrid.io.input.InputDataProvider.InputDataEvent
+import edu.ie3.osmogrid.model.{OsmoGridModel, SourceFilter}
+import edu.ie3.util.osm.model.OsmEntity.{Node, Relation, Way}
+import org.locationtech.jts.geom.{
+  Coordinate,
+  LinearRing,
+  Point,
+  Polygon,
+  PrecisionModel
+}
 
+import java.io.{FileInputStream, InputStream}
+import java.time.ZonedDateTime
+import java.util.UUID
+import scala.collection.mutable.ListBuffer
+import scala.util.{Failure, Success, Try, Using}
 import java.util.UUID
 
-object InputDataProvider extends ActorStopSupport[Request] {
+object InputDataProvider extends ActorStopSupport[InputDataEvent] {
 
+  // external requests
   sealed trait Request
 
   final case class ReqOsm(
-      runId: UUID,
       replyTo: ActorRef[InputDataProvider.Response],
-      filter: PbfFilter
+      filter: SourceFilter
   ) extends Request
+      with InputDataEvent
 
   final case class ReqAssetTypes(
       runId: UUID,
       replyTo: ActorRef[InputDataProvider.Response]
   ) extends Request
+      with InputDataEvent
 
-  object Terminate extends Request
+  case object Terminate extends Request with InputDataEvent
 
   // external responses
   sealed trait Response
-  final case class RepOsm(runId: UUID, osmModel: OsmoGridModel) extends Response
-  final case class InvalidOsmRequest(reqRunId: UUID, actualRunId: UUID)
+  final case class RepOsm(runId: UUID, osmModel: OsmoGridModel)
+      extends Response
+      with InputDataEvent
+  final case class OsmReadFailed(reason: Throwable)
+      extends Response
+      with InputDataEvent
+  final case class RepAssetTypes(assetInformation: AssetInformation)
       extends Response
```

Comment:
```suggestion
      extends Response
      with InputDataEvent
```

In case, you'd keep the `InputDataEvent`

Reply:
Well `RepAssetTypes` is _actually_ a type that is only supposed to be sent out, right?